tests/cpp/test_matmul_aten_evaluation.cpp

```diff
@@ -68,6 +68,44 @@ TEST_F(MatmulATenEvaluationTest, MmaOpAndCast) {
   EXPECT_TRUE(at::allclose(out[0], out_ref));
 }
 
+TEST_F(MatmulATenEvaluationTest, TransposeMmaOpAndCast) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  int64_t m = 32, n = 64, k = 128;
+  std::vector<int64_t> a_shape{m, k}, b_shape{n, k};
+
+  auto tv0 = makeConcreteTensor(a_shape, DataType::Half);
+  auto tv1 = makeConcreteTensor(b_shape, DataType::Half);
```

Comment:
Is there a reason to make these concrete? Actually, what are the contiguity requirements for `at::matmul` and `at::addmm`? If there are none, then it would be nice to have tests that have discontiguous inputs. Also having some with inputs that have M, K, and N not divisible by 8 would let us test cases with unaligned rows (currently not supported in nvfuser matmul).

Reply:
I think these are great points. These apply to the tests for ATen execution, and not just this test. I created a GH issue for this (https://github.com/NVIDIA/Fuser/issues/2032).